### DISCUSSION TITLE
How to make codes

### DISCUSSION THREAD

**Discussion by PerryCallie** [community/community#77816] [Programming Help]
### Select Topic Area

Question

### Body

I don't know how to build codes

**Comment by jane-does-coding** [comment]
What do you mean? You don't know how to write code or you don't know how to push it to github?

**Comment by clyde-dot** [comment]
That's completely okay! Everyone starts somewhere, and coding can seem daunting at first. There are plenty of resources available online, like tutorials, courses, and forums, that can help you get started. What specifically are you interested in coding? Maybe I can point you in the right direction!

**Comment by ghost** [comment]
Hey @PerryCallie looks like you joined github recently, as you already important step in programming, its very easy to begin you wonderful programming journey.

# Choose a Programming Language:
There are many ones like C++, C, C#, Java, Ruby, Go, Rust, Python, JS, src. But personally I would suggest Python or Javascript/JS.

# Learn the basics:
Learn concepts like,

- variables
- data types
- functions
- loops
- if/else statements
- clases
[These Differ in various Languages]

# Start building New Projects
Build some simple projects to gain practical Knowledge. You can even refer code on Github platform.

~ This where you need to delve deep into learning the language you chose ~

#  Contribute

Once you are sure that you have a decent knowledge on the domain , contribute to projects on Github.

From here learn the things you wish everyday. Don't forget sticking up with problems is very common, if you do, ask help in community discussions in github or in stackoverflow.

Happy Programming Journey
